package/src/components/MessageList/MessageList.tsx

```diff
@@ -756,20 +753,20 @@ const MessageListWithContext = (props: MessageListPropsWithContext) => {
       const showUnreadUnderlay = !!shouldShowUnreadUnderlay && showUnreadSeparator;
 
       const wrapMessageInTheme = client.userID === message.user?.id && !!myMessageTheme;
-      const renderDateSeperator = isMessageWithStylesReadByAndDateSeparator(message) &&
-        message.dateSeparator && <InlineDateSeparator date={message.dateSeparator} />;
+      const renderDateSeperator = dateSeparators[message.id] && (
+        <InlineDateSeparator date={dateSeparators[message.id]} />
+      );
       const renderMessage = (
         <Message
           goToMessage={goToMessage}
-          groupStyles={
-            isMessageWithStylesReadByAndDateSeparator(message) ? message.groupStyles : []
-          }
+          groupStyles={messageGroupStyles[message.id] ?? []}
           isTargetedMessage={highlightedMessageId === message.id}
           lastReceivedId={
             lastReceivedId === message.id || message.quoted_message_id ? lastReceivedId : undefined
           }
           message={message}
           onThreadSelect={onThreadSelect}
+          readBy={readData[message.id]}
```

Comment:
Following up on the comment above, this would allow us to:

- Not have to drill props to every `Message` component (which is costly, because it forces a relatively expensive rerender of each `FlatList` item)
- Have `renderItem` depend on less `props` (which we want, because we want it to be as pure as possible)